spec/lita/services/market_manager_spec.rb

```diff
@@ -0,0 +1,192 @@
+require 'spec_helper'
+require 'pry'
+require 'dotenv/load'
+
+describe Lita::Services::MarketManager, lita: true do
+  let(:robot) { Lita::Robot.new(registry) }
+  let(:karmanager) do
+    Lita::Services::Karmanager.new(Lita::Handlers::LunchReminder.new(robot).redis)
+  end
+  let(:lunch_assigner) do
+    Lita::Services::LunchAssigner.new(Lita::Handlers::LunchReminder.new(robot).redis, karmanager)
+  end
+  let(:subject) do
+    described_class.new(
+      Lita::Handlers::LunchReminder.new(robot).redis,
+      lunch_assigner,
+      karmanager
+    )
+  end
+  let(:fdom) { Lita::User.create(127, mention_name: 'fdom') }
+  let(:andres) { Lita::User.create(137, mention_name: 'andres') }
+  let(:oscar) { Lita::User.create(147, mention_name: 'oscar') }
+  let(:fernanda) { Lita::User.create(157, mention_name: 'fernanda') }
+
+  def add_limit_order(order_id, user, type, created_at)
+    order = {
+      id: order_id,
+      user_id: user.id,
+      type: type,
+      created_at: created_at
+    }.to_json
+    subject.add_limit_order(order)
+    order
+  end
+
+  def setup_lunchers
+    [andres, fdom, oscar].each do |user|
+      lunch_assigner.add_to_lunchers(user.mention_name)
+      lunch_assigner.add_to_current_lunchers(user.mention_name)
+      lunch_assigner.add_to_winning_lunchers(user.mention_name)
+    end
+
+    lunch_assigner.add_to_lunchers(fernanda.mention_name)
+  end
+
+  describe '#add_limit_order' do
+    context 'first order added' do
+      before do
+        @order_id = SecureRandom.uuid
+        @order_time = Time.now
+        add_limit_order(@order_id, fdom, 'sell', @order_time)
```

Comment:
Opinión personal, pero encuentro mejor que el método que se está probando (en este caso `(@order_id, fdom, 'sell', @order_time)`) se coloque en cada `it` y no en el `before`.  

Esto para que las pruebas queden más leíbles y encapsuladas, porque en el bloque `before` solo se debe colocar las condiciones "previas" para que las pruebas dentro del `it` puedan correr.

Si ahora voy directo a leer lo que está dentro del bloque `it`, no logro ver a primera vista qué estás probando.